Sources/fs/Directory.swift

```diff
@@ -9,34 +9,30 @@ import Dispatch
 import xsys
 import core
 
-public func readdir(_ path: String, cb: @escaping ( [ String ]? ) -> Void) {
+public func readdir(_ path: String, cb: @escaping ( Error?, [ String ]? ) -> Void) {
```

Comment:
Sure thing, I shall henceforth try to submit patch series as separate pull requests instead of having many commits on one branch!

... As a picky submitter myself, I feel that my nice, logically separate commits had a pretty cool split going on between the warning fix and the API change, which is now lost from history because it seems you squashed things into fewer commits instead of cherry-picking mine.  But I'll happily take most of the blame for not submitting in stages.  Mea culpa!

And thanks for merging!